src/components/Link/Item.js

```diff
@@ -59,20 +61,74 @@ export default class LinkItem extends React.Component {
   };
 
   componentDidMount() {
+    const { open } = this.state;
+    const { disableHoverEvents } = this.props;
+
     if ("ontouchstart" in document.documentElement) {
       this.setState({ touchEventsExist: true }); // eslint-disable-line
     }
+
+    if (open && disableHoverEvents && !this.hasListener) this.attachListener();
+  }
+
+  componentDidUpdate(prevProps, prevState) {
+    const { open: prevOpen } = prevState;
+    const { open } = this.state;
+    const { disableHoverEvents } = this.props;
+
+    if (!prevOpen && open && disableHoverEvents && !this.hasListener) {
+      this.attachListener();
+    } else if (prevOpen && !open && this.hasListener) {
+      this.removeListener();
+    }
   }
 
+  componentWillUnmount() {
+    if (this.hasListener) this.removeListener();
+  }
+
+  hasListener = false;
+
+  ref = this.props.disableHoverEvents ? createRef() : undefined;
```

Comment:
We should not put any logic in here, as this will only run when the component gets instantiated. And if say later the prop changes from `false` to `true` - the ref will be missing as it has not been created when instance of the component got created, and component will not work as expected (`this.ref` will be undefined in `handleOutsideClick` handler).

Reply:
Good point. I'll instantiate the `ref` from within `componentDidUpdate`, so we can respect prop changes.